pkgs/build-support/dotnet/build-dotnet-module/fetch-deps.sh

```diff
@@ -4,6 +4,8 @@ tmp=$(mktemp -d)
 trap 'chmod -R +w "$tmp" && rm -fr "$tmp"' EXIT
 
 HOME=$tmp/.home
+export TMPDIR="$tmp/.tmp"
```

Comment:
It'd be nice to restore this after the script is over so people don't have to restart their shells

Reply:
It's not really meant to be sourced in a user shell.  It's only meant to be called via nix-shell in the package-fetch-deps script.

I actually would have preferred to make it all work in one file with nix-shell shebangs, but I couldn't find a way to pass the derivation path, etc.